packit/vm_image_build.py

```diff
@@ -0,0 +1,184 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+"""
+Code that backs up Image Builder integration.
+
+Guides and more info about Image Builder:
+    https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/composing_a_customized_rhel_system_image/index
+    https://github.com/packit/research/tree/main/image-builder
+"""
+import logging
+from typing import List
+
+import requests
+
+from packit.exceptions import PackitException
+
+logger = logging.getLogger("packit")
+REDHAT_SSO_AUTH_URL = (
+    "https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token"
+)
+REDHAT_API_GET_USER_URL = "https://api.access.redhat.com/account/v1/user"
+IMAGE_BUILDER_API_URL = "https://console.redhat.com/api/image-builder/v1"
+
+
+class ImageBuilder:
+    """
+    Client class to work with Image Builder API and Red Hat SSO.
+    """
+
+    def __init__(
+        self,
+        refresh_token: str,
+    ):
+        """
+        Args:
+            refresh_token: your personal token to access Red Hat SSO, obtain at:
+                https://access.redhat.com/management/api
+        """
+        self.refresh_token = refresh_token
+
+        self.__access_token = self.__get_access_token()
```

Comment:
One thing that might confuse developers is that when they use `ImageBuilder` object for a longer time, it will stop working after the access token expires. Maybe you want to implement refreshing just in case?

Reply:
very good idea! we could even hit that ourselves

will the request that fails auth end with 403? so should we just refresh the token when that happens?

Reply:
401 but otherwise you are correct - if you get 401, just refresh the token and try again.

The token route will actually return you an `expires_in` field with the token validity duration (so you can refresh preemptively) but from our experience it's just better to rely on the service returning 401 and then do refresh.